docs/pages/upgrading/client-tools-autoupdate.mdx

```diff
@@ -0,0 +1,135 @@
+---
+title: Teleport Client Tools Automatic Updates
+description: Explains how to use Teleport client tools (`tsh` and `tctl`) auto-updates.
+---
+
+This documentation explains how to keep Teleport client tools like `tsh` and `tctl` up-to-date.
+Updates can be automatic or self-managed, ensuring tools are secure, free from bugs,
+and compatible with your Teleport cluster.
+
+## Why Keep Client Tools Updated?
+
+- **Security**: Updates deliver patches for known vulnerabilities.
+- **Bug Fixes**: Resolved issues are pushed to endpoints.
+- **Compatibility**: Avoid manual understanding of [Teleport component compatibility rules](https://goteleport.com/docs/upgrading/overview/#component-compatibility).
+
+---
+
+## Automatic Updates
+
+### How It Works
+
+When you run `tsh login`, the tool will check if updates are enabled for your cluster.
+If your client version differs from the cluster's required version, it will:
+
+1. Download the updated version.
+2. Store it in `~/.tsh/bin`.
+3. Validate the binary with a checksum.
+4. Re-execute using the updated version (with the same environment variables).
+
+### Key Features
+
+- **Binary Management**: Original binaries are preserved, and updates are stored separately.
+
+Updates are installed in the `$TELEPORT_HOME/.tsh/bin/` folder. When client tools (`tctl` or `tsh`) are executed
+from any other path, they consistently check for binaries in the update folder and re-execute them if found.
+After logging out from the cluster, the update folder is cleaned up, as each cluster might require its own version.
+
+- **Validation**: Downloaded packages are verified with a hash sum to ensure integrity.
+
+Package downloads are directed to the `cdn.teleport.dev` endpoint and depend on the operating system,
+platform, and edition. The edition must be identified by the original client tools binary.
+The URL pattern is as follows:
+- `https://cdn.teleport.dev/teleport-{ent-}vX.Y.Z-{linux,darwin,windows}-{amd64,arm64,arm,386}-{fips-}bin.tar.gz`
+- `https://cdn.teleport.dev/teleport-{ent-}vX.Y.Z-{linux,darwin,windows}-{amd64,arm64,arm,386}-{fips-}bin.tar.gz.sha256`
+
+- **Concurrency**: Tools use a locking mechanism to enable smooth operation during updates.
+
+Only one process can acquire the lock to update client tools, while other processes wait for the lock to be released.
+If the first process cancels the update, the next process in line will initiate the update.
+
+### User Experience
```

Comment:
Does this section need to be in the documentation? If the update message appears when a user executes any `tctl` or `tsh` command, the would see the update message without needing to consult the docs.

Reply:
I need to add information about cancellation. This is just an example of how to use it. Should I rename it to `Example` or move it to another section?